tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -168,6 +169,43 @@ public function testDontFinishNoCache() : void
         self::assertCount(2, $this->sqlLogger->queries);
     }
 
+    /**
+     * @param array<string, int> $params
+     * @param array<string, int> $types
+     *
+     * @dataProvider provideParamsAndTypes
+     */
+    public function testCacheFetchWithParamsAndTypes(array $params, array $types) : void
```

Comment:
Does this require functional testing? The caching layer doesn't interact with any driver- or platform-specific APIs.

Reply:
Not necessarily. I've added them to already existing functional test about executed queries and cache that didn't cover the used parameters yet.